src/components/Filters/index.tsx

```diff
@@ -0,0 +1,245 @@
+import { faFilter } from '@fortawesome/free-solid-svg-icons/faFilter';
+import { useClickOutside } from '../../hooks/useClickOutside';
+import type { JSONSchema7 as JSONSchema } from 'json-schema';
+import { isJSONSchema } from '../../AutoUI/models/helpers';
+
+const { Box, Button, useTheme, useMediaQuery } = Material;
+
+export interface FiltersView {
+	id: string;
+	eventName: string;
+	name: string;
+	filters: JSONSchema[];
+}
+
+type FilterRenderMode = 'all' | 'add' | 'search' | 'views' | 'summary';
+
+export interface FiltersProps {
+	schema: JSONSchema;
+	dark?: boolean;
+	filters?: JSONSchema[];
+	views?: FiltersView[];
+	onFiltersChange?: (filters: JSONSchema[]) => void;
+	onViewsChange?: (views: FiltersView[]) => void;
+	renderMode?: FilterRenderMode | FilterRenderMode[];
+	onSearch?: (searchTerm: string) => React.ReactElement | null;
+	showSaveView?: boolean;
+}
+
+export const Filters = ({
+	schema,
+	dark,
+	filters,
+	views,
+	onFiltersChange,
+	onViewsChange,
+	renderMode,
+	onSearch,
+	showSaveView,
+}: FiltersProps) => {
+	const { t } = useTranslation();
+	const theme = useTheme();
+	const matches = useMediaQuery(theme.breakpoints.up('sm'));
+	const [showFilterDialog, setShowFilterDialog] = React.useState(false);
+	const [showSearchDropDown, setShowSearchDropDown] = React.useState(false);
+	const [searchString, setSearchString] = React.useState<string>('');
+	const [editFilter, setEditFilter] = React.useState<JSONSchema>();
+	const [formData, setFormData] = React.useState<FormData[] | FormData>();
+	const searchBarRef = useClickOutside<HTMLDivElement>(() =>
+		setShowSearchDropDown(false),
+	);
+
+	React.useEffect(() => {
+		if (!editFilter) {
+			return;
+		}
+		if (editFilter.title === FULL_TEXT_SLUG) {
+			const description = parseFilterDescription(editFilter);
+			if (!description) {
+				return;
+			}
+			setFormData({
+				field: description.field,
+				operator: 'contains',
+				value: description.value,
+			});
+			return;
+		}
+		setFormData(
+			editFilter.anyOf
+				?.map((f) => {
+					if (!isJSONSchema(f)) {
+						return;
+					}
+					const description = parseFilterDescription(f);
+					if (!description) {
+						return;
+					}
+					return {
+						field: description.field,
+						operator: description.operator,
+						value: description.value,
+					};
+				})
+				.filter((f) => !!f) as FormData[],
+		);
+	}, [editFilter, setFormData]);
+
+	const handleFilterChange = React.useCallback(
+		(filter: JSONSchema | null | undefined) => {
+			if (!filter) {
+				return;
+			}
+			let existingFilters = filters;
+			if (editFilter && filters) {
+				existingFilters = filters.filter((f) => f.$id !== editFilter.$id);
+			}
+			onFiltersChange?.([...(existingFilters ?? []), filter]);
+		},
+		[filters, onFiltersChange, editFilter],
+	);
+
+	const handleDeleteFilter = React.useCallback(
+		(filter: JSONSchema) => {
+			const newFilters = filters?.filter((f) => f.$id !== filter.$id) ?? [];
+			onFiltersChange?.(newFilters);
```

Comment:
We skip the extra filtering step when `onFiltersChange` is nullish if we do this right?
```suggestion
			onFiltersChange?.(filters?.filter((f) => f.$id !== filter.$id) ?? []);
```